### DISCUSSION TITLE
When using MySQL with Herd, you are required to create a new db and migrate all the tables/views/data to it.

### DISCUSSION THREAD

**Discussion by lcallif** [beyondcode/herd-community#1179] [Ideas]
I propose that Herd:
  - Have the ability to point to a MySQL db created by Herd (already the one and only possibility) - Only solution - Done
        - I didn't see it documented, but being able to create a MySQL db via Herd would be a nice touch.
  - Have the ability to point to an existing MySQL db created via manager-ox's connection - Not available

As an example, I currently have a MySQL db on my MacBook Air.  The db is currently turned on/off via manager-osx.  When it is turned off, I turn on the MySQL connection in Herd, the Application fails as the "laravel" db cannot be found.  Turn off the Herd db connection, turn on the db connection via manager-osx, the application works.
  - The same issue occurs with PhpStorm.

Issue with the current approach
  - Multiple db's to be managed 
       - Make modifications in one needs to have the correspond changes done in the other.
       - Nothing is gained 
       - Additional disk space is required.  Yeah, ok, this is not a huge impact.
       - Additional maintenance.
       - Addition of one more point of failure to deal with.

Background
  - The Herd Documentation 
       - Does not state that in order to use a MySQL db that a new MySQL db is required to be created and that it needs to be created within the Herd's MySQLdb space.  
       - Does state that Herd can connect to a MySQL db.  
            - This infers that Herd can connect to an existing db.....
Notes
  - PhpStorm will connect to the MySQL database which is currently running whether by Herd or manager-osx.  Seems as if Herd should be able to connect to an existing MySQL db, whether created by Herd or not.

And yes, I understand 
 - All that is needed is to create a new db via Herd
 - Move the tables/views/data from a current MySQL db to the db created by Herd
 - Maintain both databases if/as needed 
 - manager-osx cannot start the Herd's MySQL connection
     - Not necessarily a good argument as Herd is fairly new and manager-osx has not been modified as such.  This could be for a variety of reasons, such as the Herd User-base may not be substantial enough for this to be actioned on.

I purchased Herd for a variety of reasons, but the main reason was the capability to have the MySQL db, php version selection, and web server all under 1 app.  The log feature is nice but I purchased Ray for its' extensive logging capability which Herd doesn't have (per Herd's documentation it only has dd and dump).

Thank you for taking the time to read this.

**Comment by mihas7f** [comment]
I recently purchased Laravel Herd Pro, and I'm quite disappointed to find out that the functionality to choose a database when adding a new project is not available out of the box. Considering that I need to pay extra to have an integrated database service, I feel a bit cheated. I hope the team considers adding this as a default feature in the future.

**Reply by mpociot** [reply]
When you use Laravel, the default `php artisan migrate` command will already check if a database exists and create it for you if not

**Reply by mihas7f** [reply]
Yes, I agree, but I believe the intention of the developers, as advertised on the Laravel Herd website, is "One-click PHP development environment. Zero dependencies. Zero headaches." Especially since paying for the Pro version specifically gives you access to the integrated database service within Herd. It feels like this feature should be more streamlined and built-in, aligning with the promise of simplicity and convenience.
